spec/services/events/create_batch_service_spec.rb

```diff
@@ -8,39 +8,107 @@
   let(:organization) { create(:organization) }
   let(:billable_metric) { create(:billable_metric, organization: organization) }
   let(:customer) { create(:customer, organization: organization) }
+  let(:subscription) { create(:active_subscription, customer:, organization:) }
+  let(:subscription2) { create(:active_subscription, customer:, organization:) }
+
+  before do
+    subscription
+    subscription2
+  end
 
   describe '#validate_params' do
     let(:event_arguments) do
       {
         transaction_id: SecureRandom.uuid,
-        external_subscription_ids: %w[id1 id2],
-        code: 'foo',
+        external_subscription_ids: [subscription.external_id, subscription2.external_id],
+        code: billable_metric.code,
       }
     end
 
-    it 'validates the presence of the mandatory arguments' do
-      result = create_batch_service.validate_params(params: event_arguments)
+    it 'successfully validates event arguments' do
+      result = create_batch_service.validate_params(organization:, params: event_arguments)
 
       expect(result).to be_success
     end
 
-    context 'with missing or nil arguments' do
+    context 'with missing external_subscription_ids field' do
+      let(:event_arguments) do
+        {
+          transaction_id: SecureRandom.uuid,
+          code: billable_metric.code,
+        }
+      end
+
+      it 'returns an error' do
+        result = create_batch_service.validate_params(organization:, params: event_arguments)
+
+        aggregate_failures do
+          expect(result).not_to be_success
+          expect(result.error).to be_a(BaseService::NotFoundFailure)
+          expect(result.error.error_code).to eq('subscription_not_found')
+        end
+      end
+    end
+
+    context 'with invalid external_subscription_id value' do
+      let(:event_arguments) do
+        {
+          transaction_id: SecureRandom.uuid,
+          external_subscription_ids: [subscription.external_id, subscription2.external_id, 'invalid'],
+          code: billable_metric.code,
+        }
+      end
+
+      it 'returns an error' do
+        result = create_batch_service.validate_params(organization:, params: event_arguments)
+
+        aggregate_failures do
+          expect(result).not_to be_success
+          expect(result.error).to be_a(BaseService::NotFoundFailure)
+          expect(result.error.error_code).to eq('subscription_not_found')
+        end
+      end
+    end
+
+    context 'with invalid customer reference' do
+      let(:subscription3) { create(:active_subscription) }
+      let(:event_arguments) do
+        {
+          transaction_id: SecureRandom.uuid,
+          external_subscription_ids: [subscription3.external_id, subscription2.external_id],
```

Comment:
RSpec/ScatteredLet: Group all let/let! blocks in the example group together. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ScatteredLet)